src/LongRange/KContainer.cpp

```diff
@@ -306,7 +306,7 @@ void KContainer::BuildKLists(ParticleLayout_t& lattice, bool useSphere)
   //Now fill the array that returns the index of -k when given the index of k.
   minusk.resize(numk);
   // Create a map from the hash value for each k vector to the index
-  std::map<int, int> hashToIndex;
+  std::map<long long, int> hashToIndex;
```

Comment:
instead of the overload set and all the casts in the header, since this is private and used only here what about a generic lambda?

```C++
auto getHashOfVec = [](const auto& inpv, int hashparam)->long long{
    long long hash = 0; // this will cause integral promotion below
    for(int i = 0; i < inpv.Size; ++i)
      hash += inpv[i] + hash * hashparam;
    return hash;
  };
```

Reply:
That's slick.  Doing it.